exec/java-exec/src/main/java/org/apache/drill/exec/util/StdErrCapturer.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.util;
+
+
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * The safe way to use this class is to redirect System.err to nullOutputStream.
+ * If callers want to get the redirected output -ito print or log elsewhere - as used in unit tests and they
+ * need the output to be neat, then there should be advanced ways to preserve the output to avoid concurrent 
+ * inputs messing up. But this class is not designed for that.
+ */
+public class StdErrCapturer {
```

Comment:
If an exception occurs, this class can leave the System.stderr redirected, causing mysterious errors. Suggestion: make this AutoClosable: 

try ( StdErrCapturer cap = new StdErrCapturer( ) ) {
  // Do your stuff here
}

The above will ensure that stderr is restored whether the enclosed code succeeds or fails.